README.md

```diff
@@ -11,11 +13,135 @@ The modern, distributed and scalable implementation of a game inspired by Connec
+
+## Separation of Concerns
+The service names are all a reference to the popular anime `JoJo's Bizarre Adventure`.
+
+| Name       | Role            | Description                                                                   |
+| ---------- | --------------- | ----------------------------------------------------------------------------- |
+| Doppio     | Frontend        | Svelte-based web client                                                       |
+| Speedwagon | Load balancer   | Envoy-based load balancing reverse proxy                                      |
+| Joestar    | Scaling backend | Micronaut server for user authentication and caching                          |
+| Rohan      | Central backend | Micronaut server for central storage and processing of the game and its users |
+
+### Frontend
+The frontend is written in [Svelte](https://svelte.dev/).
+As it's main purpose is to display the current state of the board, we decided that frameworks such as Angular are overkill.
+
+### Load balancing
+We use [Envoy](https://www.envoyproxy.io/) as a reverse proxy to handle load balancing.
+We had to use Envoy as it's the only reverse proxy which currently supports grpc-web.
+
+### Backend
+The backend is split into two parts:
+* Scaling: Communicates directly with our Frontend, issues and validates JWT, caches game state and sends game updates to all clients
+* Central: Manages the actual game state, synchronizes requests and handles game logic, stores persistent information, such as user credentials and scores, in a json document
+
+Both backends use [Micronaut](https://micronaut.io/) with [Kotlin](https://kotlinlang.org/).
+
+### Storage
+We decided not to use a database but instead store the (very minimalistic) data in a JSON document.
+
+## Communication
+Bidirectional communication is enabled through [gRPC](https://grpc.io/).
+For example, this allows `Rohan` to send a game update event to all `Joestar` instances, which then forward them realtime to all `Doppio` clients.
+TODO:
+- gRPC:
+  - Authentication: Not used because of lack of support by Svelte ecosystem (overlap with [Symmetric JWT](#symmetric-jwt)?)?
```

Comment:
According to [Wikipedia](url),

> gRPC supports the usage of TLS and token based authentication

.
This was not an option for us, but I don't remember why.

Reply:
We couldn't easily use the micronaut-security package because the feature is still WIP for gRPC (see https://github.com/micronaut-projects/micronaut-grpc/issues/164).
The official token based authentication works by using Google as a token provider, but we didn't want to have a vendor lock-in. 
In the end we decided to use simple symmetric tokens because of the project scope. The client does use the metadata to send the token which was recommended back in 2018 by the grpc-web team (https://github.com/grpc/grpc-web/issues/207#issuecomment-406134504)

Reply:
Done